File: BJ/Solved/11659.swift
////
////  11659.swift
////  BJ
////
////  Created by 송재훈 on 9/29/24.
////
//
//
//let input1 = readLine()!.split(separator: " ").map { Int(String($0))! }
//let (N, M) = (input1[0], input1[1])
//
//let arr = [0] + readLine()!.split(separator: " ").map { Int(String($0))! }
//var dp = Array(repeating: 0, count: 100010)
//
//for i in 1...N {
//    dp[i] = dp[i - 1] + arr[i]
//}
//
//for _ in 0..<M {
//    let range = readLine()!.split(separator: " ").map { Int(String($0))! }
//    print(dp[range[1]] - dp[range[0] - 1])
//}

File: Python/Solved/11659.py
# 5 3
# 5 4 3 2 1
# 1 3
# 2 4
# 5 5

import sys
input = sys.stdin.readline

N, M = map(int, input().split(" "))
A = list(map(int, input().split(" ")))
dp = [0] * (N + 1)
for i in range(1, N + 1): dp[i] = dp[i - 1] + A[i - 1]

for _ in range(M):
    s, e = map(int, input().split(" "))
    print(dp[e] - dp[s - 1])